smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlNodeParser.java

```diff
@@ -135,6 +135,10 @@ static ObjectNode parseObjectNode(IdlModelParser parser, String parent) {
                 String key = parseNodeObjectKey(parser);
                 parser.ws();
                 parser.expect(':');
+                if (parser.peek() == '=') {
+                    throw parser.syntax("Defining structures inline with the `:=` syntax may only be used when "
```

Comment:
This doesn't seem accurate. We're not defining a structure here, it's an object node.

Reply:
I updated the wording to be more clear